### PR TITLE
feat(compaction group): add new compaction groups in need

### PR DESCRIPTION
**Pull request by soundOfDestiny**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

add new compaction groups in need

background:

Discussions on compaction group:

* Plan to introduce more compaction groups into our system. Experiments WIP.
* Prefer using separate HummockVersion for different compaction groups even when we have more compaction groups after considering using shared L0, which introduces excessive complexity in compaction and is error-prone.
* Discuss about optimizations to handle increasing IOPS and small objects issue.

design:

Create new compaction groups on MV creation if arguments of creation hint that this MV should have separate compaction groups.

## Checklist

- [x] I have written necessary rustdoc comments

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

### CODE REVIEW

**Comment by soundOfDestiny** [comment]
will add `revert_func` later

**Comment by fuyufjh** [comment]
Can you explain the background and design?

File: src/meta/src/hummock/compaction_group/manager.rs [hunk 1]
```diff
@@ -282,14 +294,34 @@ impl CompactionGroupManagerInner {
...
-        for (table_id, compaction_group_id, table_option) in pairs {
-            let mut compaction_group = compaction_groups
-                .get_mut(*compaction_group_id)
-                .ok_or(Error::InvalidCompactionGroup(*compaction_group_id))?;
+        for (table_id, compaction_group_id, table_option) in pairs.iter_mut() {
+            let mut compaction_group =
+                if *compaction_group_id == StaticCompactionGroupId::NewCompactionGroup as u64 {
+                    current_max_id += 1;
+                    *compaction_group_id = current_max_id;
+                    compaction_groups.insert(
```
Comment (zwang28) [inline src/meta/src/hummock/compaction_group/manager.rs:312]: Need initialize a CompactStatus for this new group too.
Previously we initialize all predefined groups on system start https://github.com/risingwavelabs/risingwave/blob/91f53d8fbf614b79eadeace1499e94e7a33e4efa/src/meta/src/hummock/manager/mod.rs#L270, but now we may initialize lazily.

Comment (soundOfDestiny) [reply src/meta/src/hummock/compaction_group/manager.rs:312]: > Need initialize a CompactStatus for this new group too. Previously we initialize all predefined groups on system start
> 
> https://github.com/risingwavelabs/risingwave/blob/91f53d8fbf614b79eadeace1499e94e7a33e4efa/src/meta/src/hummock/manager/mod.rs#L270
> 
> , but now we may initialize lazily.

fixed

**Comment by soundOfDestiny** [comment]
> Can you explain the background and design?

Discussions on compaction group:

* Plan to introduce more compaction groups into our system. Experiments WIP.
* Prefer using separate HummockVersion for different compaction groups even when we have more compaction groups after considering using shared L0, which introduces excessive complexity in compaction and is error-prone.
* Discuss about optimizations to handle increasing IOPS and small objects issue.

**Review by zwang28 — APPROVED** [review]

**Pull request merged**